src/net/sourceforge/kolmafia/request/RelayRequest.java

```diff
@@ -733,6 +733,18 @@ private void sendLocalFile(final String filename) {
     } catch (IOException e) {
     }
 
+    // If it's a binary file, send it back without loading it as a string.
+    if (!this.contentType.startsWith("text/") && !this.contentType.startsWith("application/")) {
+      this.rawByteBuffer = ByteBufferUtilities.read(override);
+      if (this.rawByteBuffer.length == 0) {
+        this.sendNotFound();
+        return;
+      }
+      this.statusLine = "HTTP/1.1 200 OK";
```

Comment:
up a bit in the same file, when sending an image we set Last-Modified so the browser can send 304 if it wants; I think that is a good idea to do.